src/Generation/GirModel/Constructor.cs

```diff
@@ -8,5 +8,6 @@ public interface Constructor
     ReturnType ReturnType { get; }
     string CIdentifier { get; }
     IEnumerable<Parameter> Parameters { get; }
+    public bool Throws { get; }
```

Comment:
Idea to streamline the code a bit:

The [rnc file](https://gitlab.gnome.org/GNOME/gobject-introspection/-/blob/main/docs/gir-1.2.rnc#L501) declares an element called `Callable`. I think we should introduce an interface called `Callable` in the GirModel project to have a common base for `Method`, `Constructor` and `Function`. This interface would probably look something like:

```CS
public interface Callable
{
  string Name { get; }
  string CIdentifier { get; }
  bool Throws { get; }
  IEnumerable<Parameter> Parameters { get; }
  InstanceParameter? InstanceParameter { get; }
}
```

Then `GirModel.Function`, `GirModel.Method` and `GirModel.Constructor` could inherit this interface. The code would need to implement the new interface explicitly.

Additionally the `Method` interface would "hide" the `InstanceParameter` property:

```CS
public interface Method
{
  ...
  new InstanceParameter InstanceParameter { get; }
  ...
}
```

This allows us to be sure that methods will always have an instance parameter. `Callables` on the other hand might have an instance parameter or not.

Now you can rewrite `Generator.Renderer.Internal.Errors.Render` to accept a `Callable` and detect automatically what to do in regards to the comma.

Additionally I would create a similar renderer to the internal one for the public namespace to replace this:

```CS
if (function.Throws)
{
  string separator = parameters.Any() ? ", " : string.Empty;
  call.Append($"{separator}out var error");
}
```

with something like:
```CS
call.Append(Error.Render(function));
//OR
call.Append(Error.Render(method));
//OR
call.Append(Error.Render(constructor));
```

Reply:
That sounds like a good idea 👍 , it did feel like there was a bit too much duplicate code between the different cases

Reply:
Updated with those changes - I think it definitely works better, thanks for the suggestions!